src/Shared/FileMatcher.cs

```diff
@@ -1441,7 +1443,11 @@ enum GetSearchDataResult
                 needsRecursion);
 
             result.SearchData = searchData;
-            result.BaseDirectory = fixedDirectoryPart;
+
+            // The double trim fixes https://github.com/Microsoft/msbuild/issues/917
+            // System.IO does not leave trailing slashes in directory enumerations.
+            // Since this string will be matched against System.IO directory enumerations, both need to agree on trailing slashes
+            result.BaseDirectory = fixedDirectoryPart.TrimEnd(Path.AltDirectorySeparatorChar).TrimEnd(Path.DirectorySeparatorChar);
```

Comment:
You could do this with a single call: `.TrimEnd(Path.AltDirectorySeparatorChar, Path.DirectorySeparatorChar)`